cms/src/test/java/uk/nhs/digital/ps/modules/searchableFlagModule/JcrProvider.java

```diff
@@ -0,0 +1,85 @@
+package uk.nhs.digital.ps.modules.searchableFlagModule;
+
+import com.sun.org.apache.xpath.internal.operations.Bool;
+import org.apache.sling.testing.mock.jcr.MockJcr;
+import org.junit.Ignore;
+import org.yaml.snakeyaml.TypeDescription;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+
+import javax.jcr.Node;
+import javax.jcr.Repository;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+public class JcrProvider {
+
+    public Session getJcrFromFixture(InputStream fixture) {
+        Repository repository = null;
+        Session session = null;
+        try {
+            repository = MockJcr.newRepository();
+            session = repository.login();
+            Node rootNode = session.getRootNode();
+
+            Yaml parser = new Yaml();
+            List<LinkedHashMap<String, Object>> nodes = (ArrayList<LinkedHashMap<String, Object>>) parser.load(fixture);
+
+            nodes.forEach((object) -> {
+                NodeYaml node = new NodeYaml(
+                    (String) object.get("path"),
+                    (String) object.get("primaryType"),
+                    (LinkedHashMap<String, Object>) object.getOrDefault("properties", null)
+                );
+                try {
+                    Node n = rootNode.addNode(node.path, node.primaryType);
+                    node.properties.forEach((key, value) -> {
+                        try {
+                            if (value instanceof Boolean) {
+                                n.setProperty(key, (Boolean) value);
+                            } else {
+                                n.setProperty(key, value.toString());
+                            }
+                        } catch (RepositoryException e) {
+                            e.printStackTrace();
+                        }
+                    });
+                } catch (RepositoryException e) {
+                    e.printStackTrace();
+                }
+            });
+
+            session.save();
+        } catch (RepositoryException e) {
+            e.printStackTrace();
+        }
+
+        return session;
+    }
+
+    public static class NodeYaml {
+        public String path;
+        public String primaryType;
+        public LinkedHashMap<String, Object> properties;
+
+        public NodeYaml(String path, String primaryType, LinkedHashMap<String, Object> properties) {
+            this.path = path;
+            this.primaryType = primaryType;
+            if (properties == null) {
```

Comment:
can pass `emptyMap()` into the `object.getOrDefault()` method above instead of null, then you don't need this extra null check